src/pyhf/workspace.py

```diff
@@ -17,7 +17,7 @@
 log = logging.getLogger(__name__)
 
 
-def _join_items(join, left_items, right_items, key='name'):
+def _join_items(join, left_items, right_items, key='name', deep_merge_key=''):
```

Comment:
```suggestion
def _join_items(join, left_items, right_items, key='name', deep_merge_key=None):
```

Probably keep this as `None` instead of empty string?